qos-monitor/src/main/java/eu/arrowhead/core/qos/service/QoSMonitorDriver.java

```diff
@@ -257,4 +440,92 @@ private UriComponents getGatekeeperInitRelayTestUri() {
 		
 		throw new ArrowheadException("QoS Monitor can't find gatekeeper init_relay_test URI.");
 	}
+
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents getOrchestrationProcessUri() {
+		logger.debug("getOrchestrationProcessUri started...");
+
+		if (arrowheadContext.containsKey(ORCHESTRATION_PROCESS_URI_KEY)) {
+			try {
+				return (UriComponents) arrowheadContext.get(ORCHESTRATION_PROCESS_URI_KEY);
+			} catch (final ClassCastException ex) {
+				throw new ArrowheadException("QoS Monitor can't find orchestration process URI.");
+			}
+		}
+
+		throw new ArrowheadException("QoS Monitor can't find orchestration process URI.");
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents getEventHandlerSubscribeUri() {
+		logger.debug("getEventHandlerSubscribeUri started...");
+
+		if (arrowheadContext.containsKey(EVENT_SUBSCRIBE_URI_KEY)) {
+			try {
+				return (UriComponents) arrowheadContext.get(EVENT_SUBSCRIBE_URI_KEY);
+			} catch (final ClassCastException ex) {
+				throw new ArrowheadException("QoS Monitor can't find Event Handler subscribe URI.");
+			}
+		}
+
+		throw new ArrowheadException("QoS Monitor can't find Event Handler subscribe URI.");
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents getEventHandlerUnsubscribeUri() {
+		logger.debug("getEventHandlerUnsubscribeUri started...");
+
+		if (arrowheadContext.containsKey(EVENT_UNSUBSCRIBE_URI_KEY)) {
+			try {
+				return (UriComponents) arrowheadContext.get(EVENT_UNSUBSCRIBE_URI_KEY);
+			} catch (final ClassCastException ex) {
+				throw new ArrowheadException("QoS Monitor can't find Event Handler unsubscribe URI.");
+			}
+		}
+
+		throw new ArrowheadException("QoS Monitor can't find Event Handler unsubscribe URI.");
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private SubscriptionRequestDTO getSubscriptionRequestDTOForExternalPingMonitor(final SystemRequestDTO provider) {
+		logger.debug("getSubscriptionRequestDTOForExternalPingMonitor started...");
+
+		final SubscriptionRequestDTO requestTemplate = new SubscriptionRequestDTO();
+		requestTemplate.setMatchMetaData(false);
+		requestTemplate.setNotifyUri(CommonConstants.QOSMONITOR_URI + QosMonitorConstants.EXTERNAL_PING_MONITOR_EVENT_NOTIFICATION_URI);
+		requestTemplate.setSubscriberSystem(getQosMonitorSystemRequestDTO());//requester);
+	
+		if(provider != null) {
+			requestTemplate.setSources(Set.of(provider));
+		}
+
+		return requestTemplate;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private SystemRequestDTO getQosMonitorSystemRequestDTO() {
+		logger.debug("getQosMonitorSystemRequestDTO started...");
```

Comment:
system request dto should be stored in a member using lazy initialization, and after the first call just returned the member.

Reply:
done